azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py

```diff
@@ -28,20 +28,20 @@ class MQTTTransportStage(PipelineStage):
     """
 
     @pipeline_thread.runs_on_pipeline_thread
-    def _cancel_active_connect_disconnect_ops(self):
+    def _cancel_pending_connection_op(self):
```

Comment:
There is some overlap between the code in this layer and the code in my latest PR.  With my latest PR, there should never be 2 pending connection ops issued from above.  
Regardless, I think this code is still OK to have here.  Maybe we can remove it at some point, but I  think we're better off leaving the extra protection right now.

Reply:
Yeah, I think from the perspective of this module, there could be 2 pending ops, so it's important to have this logic that cancels anything already existing before setting a new one. Just to be safe.

---
In reply to: [311646559](https://github.com/Azure/azure-iot-sdk-python-preview/pull/159#discussion_r311646559) [](ancestors = 311646559)